packages/shorebird_cli/lib/src/cache.dart

```diff
@@ -58,7 +59,15 @@ class Cache {
         continue;
       }
 
-      await artifact.update();
+      await retry(
+        artifact.update,
+        maxAttempts: 3,
+        onRetry: (e) {
+          logger
+            ..err('Failed to update ${artifact.name}, retrying...')
```

Comment:
We probably want this to remain detail, no? If subsequent attempts succeed, I worry this will only confuse users.